common/system/system.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package system
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/util"
+	"github.com/hyperledger/fabric/core/middleware"
+)
+
+//go:generate counterfeiter -o fakes/logger.go -fake-name Logger . Logger
+
+type Logger interface {
+	Warn(args ...interface{})
+	Warnf(template string, args ...interface{})
+}
+
+type Options struct {
+	Logger        Logger
+	ListenAddress string
+	TLS           TLS
+}
+
+type System struct {
+	logger     Logger
+	options    Options
+	httpServer *http.Server
+	mux        *http.ServeMux
+	addr       string
+}
+
+func NewSystem(o Options) *System {
+	logger := o.Logger
+	if logger == nil {
+		logger = flogging.MustGetLogger("system.runner")
```

Comment:
Why is this `system.runner` instead of simply the package name? What value does `runner` add?

Reply:
I was following what's there in the operations system code. I'm not opposed to renaming this.

Reply:
Renamed.